pkg/network/vmispec/network.go

```diff
@@ -0,0 +1,32 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package vmispec
+
+import v1 "kubevirt.io/api/core/v1"
+
+func lookupPodNetwork(networks []v1.Network) *v1.Network {
+	for _, network := range networks {
+		if network.Pod != nil {
+			net := network
+			return &net
```

Comment:
nit: I think `return &network` is enough, `net` is redundant here

Reply:
Referencing the address of a value in a for-loop-range is problematic and several static analysis tools will warn on it.
It has been handled already in [PR#4751](https://github.com/kubevirt/kubevirt/pull/4751).

I think that if we just reference it on return, there is no problem. However, any future changes here may miss this and there is a chance that the static analysis tool will warn on this.
I think the original idea was not to ever reference the value, avoiding the need to think about this again.

Reply:
TBH I had no idea golang works that way. It amazes me such a trivial thing can cause such unexpected result. Isn't is obvious that taking the pointer of a loop element means taking the pointer of the actual element and not the iterator?.. Disappointing.

Thanks for educating me!